### PR TITLE
Correct handling of conditional phases

### LINKED ISSUE #576: `squash_1qb_to_tk1` does not preserve conditional phase
The following code

```
Circuit circ(1, 1);
circ.add_conditional_gate<unsigned>(OpType::Rz, {2.}, {0}, {0}, 1);
std::cout << circ;
Transforms::squash_1qb_to_pqp(OpType::Rz, OpType::Ry).apply(circ);
std::cout << circ;
```

prints an empty circuit with phase 1.0, ie:
```
Phase (in half-turns): 1.0
```

expected:
```
IF ([c[0]] == 1) THEN Rz(2) q[0];
Phase (in half-turns): 0.0
```

or equivalent.

### PR DESCRIPTION
**Pull request by cqc-alec**
Fixes #576 .
After merging I plan to make a dev release and then update all the extensions to make sure they work with this change,

### CODE REVIEW

File: tket/src/Transformations/Rebase.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ static bool standard_rebase(
         tk1_replacement) {
   bool success = false;
   VertexSet bin;
-  BGL_FORALL_VERTICES(v, circ.dag, DAG) {
+  for (const Vertex& v : circ.vertices_in_order()) {
```
Comment (yao-cqc) [inline tket/src/Transformations/Rebase.cpp:62]: Why do we need these changes?

Comment (cqc-alec) [reply tket/src/Transformations/Rebase.cpp:62]: Without this change, a rebase pass operating on a circuit with a conditional operation enters an infinite loop, as the vertices in the DAG are modified inside the iteration. I don't quite get why we didn't see this before, maybe just luck...?

Comment (yao-cqc) [reply tket/src/Transformations/Rebase.cpp:62]: That's interesting... anyway, the code looks safer now. However, do we really need a sorted list?

Comment (cqc-alec) [reply tket/src/Transformations/Rebase.cpp]: Done.

**Comment by cqc-alec** [comment]
> Should we also change the QASM converter to handle `OpType.Phase`s?

Yes if it doesn't already then we should. Can we do that in another PR?

**Review by yao-cqc — APPROVED** [review]

**Review by lmondada — COMMENTED** [review]
Nice work, thank you!

**Pull request merged**